tests/tck/features/optimizer/PushFilterDownHashInnerJoinRule.feature

```diff
@@ -0,0 +1,205 @@
+# Copyright (c) 2022 vesoft inc. All rights reserved.
+#
+# This source code is licensed under Apache 2.0 License.
+Feature: Push Filter down HashInnerJoin rule
+
+  Background:
+    Given a graph with space named "nba"
+
+  Scenario: push filter down HashInnerJoin
+    When profiling query:
+      """
+      MATCH (v:player)
+      WHERE id(v) == 'Tim Duncan'
+      WITH v AS v1
```

Comment:
Add some test cases related to pattern expression filter, such as:
```
 match p=(a:player)-[e:like*1..3]->(b) where b.player.age>42 
 with relationships(p)[1] AS e1 match (b)-[:serve]->(c) where c.team.name>"S" and (b)-[e1]->() 
 return c
``

Reply:
I missed this, maybe we can add this test later when optimizing the filter push-down of the RollUpApply. 
Just ignore this comment.